libs/akita-ng-effects/src/lib/akita-ng-effects.module.ts

```diff
@@ -0,0 +1,60 @@
+import { Injector, ModuleWithProviders, NgModule, Type } from '@angular/core';
+import { _FEATURE_EFFECTS, _ROOT_EFFECTS, FEATURE_EFFECT_INSTANCES, ROOT_EFFECT_INSTANCES } from './tokens';
+import { EffectsRootModule } from './effect-root.module';
+import { Actions } from './actions';
+import { EffectsFeatureModule } from './effect-feature.module';
+import { ModuleManager } from './module-manager.service';
+
+@NgModule({})
+export class AkitaNgEffectsModule {
+  static forRoot(rootEffects: Type<any>[] = []): ModuleWithProviders<EffectsRootModule> {
+    return {
+      ngModule: EffectsRootModule,
+      providers: [
+        ModuleManager,
+        Actions,
+        rootEffects,
+        {
+          provide: _ROOT_EFFECTS,
+          useValue: rootEffects
+        },
+        {
+          provide: ROOT_EFFECT_INSTANCES,
+          useFactory: createEffects,
+          deps: [Injector, _ROOT_EFFECTS]
+        }
+      ]
+    };
+  }
+
+  static forFeature(featureEffects: Type<any>[] = []): ModuleWithProviders<EffectsFeatureModule> {
+    return {
+      ngModule: EffectsFeatureModule,
+      providers: [
+        featureEffects,
+        {
+          provide: _FEATURE_EFFECTS,
+          useValue: featureEffects
+        },
+        {
+          provide: FEATURE_EFFECT_INSTANCES,
+          useFactory: createEffects,
+          deps: [Injector, _FEATURE_EFFECTS]
+        }
+      ]
+    };
+  }
+}
+
+export function createEffects(injector: Injector, effects: Type<any>[]): any[] {
+  const mergedEffects: Type<any>[] = [];
```

Comment:
Why we need `mergedEffects`? We can just call `createEffectInstances(injector, effects)`

Reply:
Absolutely correct. That was a leftover from an earlier different approach.